### PR TITLE
Dispose fixes + misc touchups

### PR DESCRIPTION
**Pull request by RoosterDragon**
- Fix some Dispose implementations, and add some missing calls to Dispose.
- Fix serialization of custom exceptions.
- Rethrow exceptions correctly.

### CODE REVIEW

File: OpenRA.Mods.Common/LoadScreens/LogoStripeLoadScreen.cs [hunk 1]
```diff
@@ -79,12 +79,15 @@ public override void Display()
 			r.EndFrame(new NullInputHandler());
 		}
 
-		public override void Dispose()
+		protected override void Dispose(bool disposing)
 		{
-			if (sheet != null)
-				sheet.Dispose();
+			if (disposing)
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/LoadScreens/LogoStripeLoadScreen.cs]: Just combine the two ifs?
`if (disposing && sheet != null)`

Comment (RoosterDragon) [inline OpenRA.Mods.Common/LoadScreens/LogoStripeLoadScreen.cs]: I tend to format it like this so future people can just dunk any extra stuff into the section without redoing the whole block, e.g.:
```
if (disposing)
{
    if (a != null)
        a.Dispose();
    b.Dispose();
}

Comment (Phrohdoh) [inline OpenRA.Mods.Common/LoadScreens/LogoStripeLoadScreen.cs]: Well there is no `a` vs `b` to worry about here, and this is inconsistent with the rest of our formatting.

Comment (RoosterDragon) [inline OpenRA.Mods.Common/LoadScreens/LogoStripeLoadScreen.cs]: OK, changed.

**Comment by Mailaender** [comment]
Something isn't right with the voxel rendering here:

image

**Comment by RoosterDragon** [comment]
So `VoxelRenderer` is actually reusing sheets and disposing them too. The extra dispose I added would mean I'd kill off textures which were then re-used next frame and hence missing. It's hacky but I can see why it's doing it that way.

I've simply removed my changes to disposing in `VoxelRenderer` instead.

**Comment by Mailaender** [comment]
:white_check_mark: No further regressions spotted.

**Comment by pchote** [comment]
:+1:

**Pull request merged**